model-providers/openai/azure-openai/runtime/src/main/java/io/quarkiverse/langchain4j/azure/openai/AzureModelAuthProviderFilter.java

```diff
@@ -0,0 +1,88 @@
+package io.quarkiverse.langchain4j.azure.openai;
+
+import java.net.URI;
+import java.util.concurrent.Executor;
+
+import jakarta.ws.rs.core.MultivaluedMap;
+
+import org.eclipse.microprofile.context.ManagedExecutor;
+import org.jboss.resteasy.reactive.client.spi.ResteasyReactiveClientRequestContext;
+import org.jboss.resteasy.reactive.client.spi.ResteasyReactiveClientRequestFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.identity.DefaultAzureCredential;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+
+import io.quarkiverse.langchain4j.auth.ModelAuthProvider;
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.InstanceHandle;
+import io.smallrye.mutiny.infrastructure.Infrastructure;
+
+public class AzureModelAuthProviderFilter implements ResteasyReactiveClientRequestFilter {
+    private static final Logger log = LoggerFactory.getLogger(AzureModelAuthProviderFilter.class);
+    private final ModelAuthProvider authorizer;
+
+    public AzureModelAuthProviderFilter() {
+        this.authorizer = new ApplicationDefaultAuthProvider();
```

Comment:
We should make sure it is loaded only when no other custom model auth providers are available, for example, I linked earlier to the demo where an access token acquired after an Entra ID authorization code flow completion is propagated